src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -99,7 +99,12 @@ public void NavigateToSourceGeneratedFile(Project project, ISourceGenerator gene
             // The file name we generate here is chosen to match the compiler's choice, so the debugger can recognize the files should match.
             // This can only be changed if the compiler changes the algorithm as well.
             var temporaryFilePath = Path.Combine(projectDirectory, $"{generatorType.Module.ModuleVersionId}_{generatorType.FullName}_{generatedSourceHintName}");
-            File.WriteAllText(temporaryFilePath, "");
+
+            // Don't write to the file if it's already there, as that potentially triggers a file reload
+            if (!File.Exists(temporaryFilePath))
```

Comment:
just want to check - the temp file could exist because it was already navigated to once, is that the scenario?

Reply:
@dibarbet Yes; if the file was already open we were still writing it a second time which triggered a reload.